apis/compute/v1beta1/router_types.go

```diff
@@ -0,0 +1,425 @@
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RouterParameters define the desired state of a Google Compute Engine
+// Router. Most fields map directly to a Router:
+// https://cloud.google.com/compute/docs/reference/rest/v1/routers/
+type RouterParameters struct {
+	// Description: An optional description of this resource. Provide this
+	// field when you create the resource.
+	// +optional
+	// +immutable
+	Description *string `json:"description,omitempty"`
+
+	// Region: URL of the region where the Subnetwork resides. This field
+	// can be set only at resource creation time.
+	// +immutable
+	Region string `json:"region"`
+
+	// Network: URI of the network to which this router belongs.
+	// +immutable
+	// +optional
+	Network *string `json:"network,omitempty"`
+
+	// NetworkRef references a Network and retrieves its URI
+	// +optional
+	// +immutable
+	NetworkRef *xpv1.Reference `json:"networkRef,omitempty"`
+
+	// NetworkSelector selects a reference to a Network
+	// +optional
+	// +immutable
+	NetworkSelector *xpv1.Selector `json:"networkSelector,omitempty"`
+
+	// Bgp: BGP information specific to this router.
+	// +optional
+	Bgp *RouterBgp `json:"bgp,omitempty"`
+
+	// BgpPeers: BGP information that must be configured into the routing
+	// stack to establish BGP peering. This information must specify the
+	// peer ASN and either the interface name, IP address, or peer IP
+	// address. Please refer to RFC4273.
+	// +optional
+	BgpPeers []*RouterBgpPeer `json:"bgpPeers,omitempty"`
+
+	// EncryptedInterconnectRouter: Field to indicate if a router is
+	// dedicated to use with encrypted Interconnect Attachment
+	// (IPsec-encrypted Cloud Interconnect feature).
+	// Not currently available in all Interconnect locations.
+	// +optional
+	EncryptedInterconnectRouter *bool `json:"encryptedInterconnectRouter,omitempty"`
+
+	// Interfaces: Router interfaces. Each interface requires either one
+	// linked resource, (for example, linkedVpnTunnel), or IP address and IP
+	// address range (for example, ipRange), or both.
+	// +optional
+	Interfaces []*RouterInterface `json:"interfaces,omitempty"`
+
+	// Nats: A list of NAT services created in this router.
+	// +optional
+	Nats []*RouterNat `json:"nats,omitempty"`
+}
+
+// A RouterBgp represents the Bgp information for router.
+type RouterBgp struct {
+	// AdvertiseMode: User-specified flag to indicate which mode to use for
+	// advertisement. The options are DEFAULT or CUSTOM.
+	//
+	// Possible values:
+	//   "CUSTOM"
+	//   "DEFAULT"
+	// +optional
```

Comment:
You might consider adding a kubebuilder enum annotation like: https://github.com/crossplane/provider-gcp/blob/80525f79de7620594a3794693c7795efec2a0a1f/apis/compute/v1beta1/globaladdress_types.go#L110